debezium-core/src/main/java/io/debezium/pipeline/metrics/PipelineMetrics.java

```diff
@@ -49,26 +53,43 @@
     }
 
     @Override
-    public void onEvent(DataCollectionId source, OffsetContext offset, Object key, Struct value) {
-        updateCommonEventMetrics();
+    public void onEvent(DataCollectionId source, OffsetContext offset, Object key, Struct value, Operation operation) {
+        updateCommonEventMetrics(operation);
         lastEvent = metadataProvider.toSummaryString(source, offset, key, value);
     }
 
-    private void updateCommonEventMetrics() {
+    private void updateCommonEventMetrics(Operation operation) {
         totalNumberOfEventsSeen.incrementAndGet();
         lastEventTimestamp.set(clock.currentTimeInMillis());
+        if (operation != null) {
+            switch (operation) {
+                case READ:
+                case CREATE:
+                    totalNumberOfCreateEventsSeen.incrementAndGet();
+                    break;
+                case UPDATE:
+                    totalNumberOfUpdateEventsSeen.incrementAndGet();
+                    break;
+                case DELETE:
+                case TRUNCATE:
```

Comment:
@jpechane, wdyt about mixing delete and truncate counters? It feels a bit unclean, OTOH the number of truncates should neglectible in comparison to deletes.